clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -560,13 +561,23 @@ private void handleProduceResponse(ClientResponse response, Map<TopicPartition,
             log.trace("Received produce response from node {} with correlation id {}", response.destination(), correlationId);
             // if we have a response, parse it
             if (response.hasResponse()) {
+                // TODO: Sender should exercise PartitionProduceResponse rather than ProduceResponse.PartitionResponse
```

Comment:
Is the plan to save this for a follow-up? It looks like it will be a bit of effort to trace down all the uses, but seems doable.

Reply:
https://issues.apache.org/jira/browse/KAFKA-10696